lib/game_engine.rb

```diff
@@ -0,0 +1,45 @@
+require_relative 'user_interface'
+class GameEngine
+  include UserInterface
+  attr_reader :playing_board, :player1, :player2
+
+  def initialize(board, player1, player2)
+    @playing_board = board
+    @player1 = player1
+    @player2 = player2
+  end
+
+  def play
+    current_player = player1
+    loop do
+      UserInterface.display_which_player_turn(current_player)
+      move = current_player.make_move(playing_board)
+      next if move.nil?
+
+      playing_board.apply_move(current_player.symbol, move)
+
+      return find_board_status(current_player) unless find_board_status(current_player).nil?
+
+      current_player = switch_players(current_player)
+    end
+  end
+
+  def find_board_status(current_player)
```

Comment:
I wonder if this is really more of the game status than the board status?

Reply:
Resolved at aa0317b6aca80d49c8de0caf1a7baf64463c8a5a